src/client.rs

```diff
@@ -250,6 +253,36 @@ impl Client {
     {
         Api::namespaced(self.client.clone(), namespace)
     }
+
+    /// Waits until given resource with metadata is recognized as applied by Kubernetes API
+    ///
+    /// # Arguments
+    ///
+    /// - `namespace` - Optional namespace to look for the resources in.
+    /// - `lp` - Parameters to filter resources in given namespace.
+    pub async fn wait_ready<T>(&self, namespace: Option<String>, lp: ListParams)
+    where
+        T: Meta + Clone + DeserializeOwned + Send + 'static,
+    {
+        let api: Api<T> = self.get_api(namespace);
+        let mut watcher = kube_runtime::watcher(api, lp).boxed();
+
+        while let Some(result) = watcher.next().await {
+            match result {
+                Ok(event) => match event {
+                    Event::Applied(_) | Event::Restarted(_) => {
```

Comment:
`Restarted` means the watch stream was restarted, no?
That in itself doesn't carry much information about whether the object is ready or now.
I'd have assumed that we only need to wait for `Applied`?

Reply:
We don't send an individual `Applied` event if it is already covered by the `Restarted`. So the correct pattern here would be something like `Event::Applied(_) | Event::Restarted([_, ..])`.

Reply:
Looks like `Event::Restarted` carries a Vec. `Event::Restarted([_, ..]` results in the following error:

```
error[E0529]: expected an array or slice, found `std::vec::Vec<T>`
   --> src/client.rs:273:58
    |
273 |                     Event::Applied(_) | Event::Restarted([_, ..]) => {
    |                                                          ^^^^^^^ pattern cannot match with input type `std::vec::Vec<T>`
```

Reply:
I wonder `restarted` can give me an empty Vec provided the `ListParams` matches nothing. I'll write a test for that.

Reply:
The necessity of catching the `restarted` event is nicely demonstrated by this test: https://github.com/stackabletech/operator-rs/pull/79/files#diff-7f93c4e263c4e9ec748f804c7fd04a3b2fde86ffd741fb5516d67e1097bae4c1R366

if the resource is already created and there are no changes, it might come to us as `restarted.

Reply:
The answer is yet, restarted can give empty Vec if ListParams filters nothing. Fixed in the code and added test for that.